File: test/leetcode/m316/RemoveDuplicateLettersTest.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m316;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Test for {@link RemoveDuplicateLetters}
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/09/14 09:20
 */
class RemoveDuplicateLettersTest {
    private RemoveDuplicateLetters test;

    @BeforeEach
    void setUp() {
        test = new RemoveDuplicateLetters();
    }

    @Test
    void test1() {
        String s = "bcabc";
        String actual = "abc";
        assertEquals(actual, test.removeDuplicateLetters(s));
    }

    @Test
    void test2() {
        String s = "cbacdcbc";
        String actual = "acdb";
        assertEquals(actual, test.removeDuplicateLetters(s));
    }
}

File: src/leetcode/m316/RemoveDuplicateLetters_test.go
package m316

import "testing"

func Test_removeDuplicateLetters(t *testing.T) {
	tests := []struct {
		name string
		s    string
		want string
	}{
		{"Test case 1", "bcabc", "abc"},
		{"Test case 2", "cbacdcbc", "acdb"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := removeDuplicateLetters(tt.s); got != tt.want {
				t.Errorf("removeDuplicateLetters() = %v, want %v", got, tt.want)
			}
		})
	}
}
